### ISSUE TITLE
Mangohud stops working in steam-session after switching session

### ISSUE THREAD

**Issue by pongo1231** [Jovian-Experiments/Jovian-NixOS#158]
I get the following in the journal when switching from steam-session to desktop session and back to steam-session session:
```
jovian-session[90519]: Failed to start transient service unit: Unit steam-session.mangoapp.service was already loaded or has a fragment file.
mangoapp[82192]: XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":0"
mangoapp[82192]:       after 55784 requests (55784 known processed) with 30 events remaining.
mangoapp[82192]: XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":0"
mangoapp[82192]:       after 352 requests (352 known processed) with 17 events remaining.
kernel: traps: mangoapp[82374] general protection fault ip:7fb8f75cfdfe sp:7fb8c6dfb7c0 error:0 in libc.so.6[7fb8f755c000+159000]
systemd-coredump[90805]: Process 82192 (mangoapp) of user 1000 dumped core.
                                                    Module mangoapp without build-id.
                                                    #1  0x000000000044f862 parse_overlay_config (mangoapp + 0x50862)
                                                    #2  0x0000000000413d59 _ZL11ctrl_threadv (mangoapp + 0x14d59)
                                                    #2  0x0000000000411e4b main (mangoapp + 0x12e4b)
                                                    #5  0x0000000000413b15 _start (mangoapp + 0x14b15)
                                                    #2  0x000000000048a2c1 _ZL11fileChangedP13notify_thread (mangoapp + 0x8b2c1)
                                                    #8  0x00000000004ae054 _ZNSt10unique_ptrI9_XDisplaySt8functionIFvPS0_EEED1Ev (mangoapp + 0xaf054)
                                                    #15 0x00000000004141b3 _ZL15msg_read_threadv (mangoapp + 0x151b3)
                                                    #3  0x00000000004a07de _Z29amdgpu_metrics_polling_threadv (mangoapp + 0xa17de)
░░ Subject: Process 82192 (mangoapp) dumped core
░░ Process 82192 (mangoapp) crashed and dumped core.
systemd[1567]: steam-session.mangoapp.service: Main process exited, code=dumped, status=11/SEGV
systemd[1567]: steam-session.mangoapp.service: Failed with result 'core-dump'.
systemd[1567]: steam-session.mangoapp.service: Consumed 8.167s CPU time.
systemd[1567]: steam-session.mangoapp.service: Scheduled restart job, restart counter is at 1.
systemd[1567]: Stopped /nix/store/j6i7qcvrfapllhlj0vpr9y6l52x5lpph-mangohud-unstable-2023-04-23/bin/mangoapp.
systemd[1567]: steam-session.mangoapp.service: Consumed 8.167s CPU time.
systemd[1567]: Started /nix/store/j6i7qcvrfapllhlj0vpr9y6l52x5lpph-mangohud-unstable-2023-04-23/bin/mangoapp.
```
Also afterwards this shows up persistently:
```
mangoapp[90852]: [2023-09-02 13:49:52.914] [error] [overlay_params.cpp:743] Unknown option 'nis_steam_sharpness'
mangoapp[90852]: [2023-09-02 13:49:52.914] [error] [overlay_params.cpp:105] Couldn't create socket pipe at 'mangohud'
mangoapp[90852]: [2023-09-02 13:49:52.914] [error] [overlay_params.cpp:106] ERROR: 'Address already in use'
```

**To replicate:**
1. Enable steam-session, autoStart and desktopSession
I use the following:
```
jovian = {
  steam = {
    enable = true;
    autoStart = true;
    user = "pongo";
    desktopSession = "plasma";
  };
};
```
2. Inside steam-session use the "Switch To Desktop" option
3. Log out to switch back to steam-session
4. Launch any game
5. Try to use the "Performance Overlay Level" option in the quick menu, it won't do anything

Restarting greetd.service fixes it.

**Comment by toast003** [comment]
Have mostly the same config as @pongo1231 except that I have `jovian.steam.desktopSession ` set to `plasmawayland`, and I have the same problem, although I have noticed mangoapp never crashes , it keeps running (even after switching to desktop from steam) but it just doesn't show up. 
Restarting greetd doesn't fix it either, only way I have seen to make it work again by stopping steam-session.mangoapp.service when on KDE

EDIT: I'm on nixpkgs `3a2786eea085f040a66ecde1bc3ddc7099f6dbeb` and jovian `d31a25e74b1bb7df342c0a53eeac4d4f91ea22b1` set to follow the system nixpkgs

**Comment by pongo1231** [comment]
Issue doesn't occur for me anymore after #167 was merged
